scene/gui/tree.cpp

```diff
@@ -99,39 +99,45 @@ void TreeItem::_change_tree(Tree *p_tree) {
 		c = c->next;
 	}
 
-	if (tree && tree->root == this) {
-		tree->root = nullptr;
-	}
+	if (tree) {
+		if (tree->root == this) {
+			tree->root = nullptr;
+		}
 
-	if (tree && tree->popup_edited_item == this) {
-		tree->popup_edited_item = nullptr;
-		tree->pressing_for_editor = false;
-	}
+		if (tree->popup_edited_item == this) {
+			tree->popup_edited_item = nullptr;
+			tree->pressing_for_editor = false;
+		}
 
-	if (tree && tree->cache.hover_item == this) {
-		tree->cache.hover_item = nullptr;
-	}
+		if (tree->cache.hover_item == this) {
+			tree->cache.hover_item = nullptr;
+		}
 
-	if (tree && tree->selected_item == this) {
-		tree->selected_item = nullptr;
-	}
+		if (tree->selected_item == this) {
+			tree->selected_item = nullptr;
+		}
 
-	if (tree && tree->drop_mode_over == this) {
-		tree->drop_mode_over = nullptr;
-	}
+		if (tree->drop_mode_over == this) {
+			tree->drop_mode_over = nullptr;
+		}
 
-	if (tree && tree->single_select_defer == this) {
-		tree->single_select_defer = nullptr;
-	}
+		if (tree->single_select_defer == this) {
+			tree->single_select_defer = nullptr;
+		}
+
+		if (tree->edited_item == this) {
+			tree->edited_item = nullptr;
+			tree->pressing_for_editor = false;
+		}
 
-	if (tree && tree->edited_item == this) {
-		tree->edited_item = nullptr;
-		tree->pressing_for_editor = false;
+		tree->update();
 	}
 
+
 	tree = p_tree;
 
 	if (tree) {
+		tree->update();
```

Comment:
Now that _change_tree calls update(), can you replace

```c++
if (old_tree && old_tree != tree) {
	old_tree->update();
}

if (tree) {
	tree->update();
}
```
with 
```c++
if (tree && old_tree == tree) {
	tree->update();
}
```
in TreeItem::move_before and TreeItem::move_after?

Reply:
Actually, `_change_tree()` will update both trees, so this can be even removed.

Reply:
No, it's necessary because _change_tree does nothing if the source and destination tree are the same.

Reply:
Right 🙃